### PR TITLE
Import Dotnet GenAPI source for local management

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Since "GenAPI" is an internally managed tool we depend on, that isn't released and does not seem planned for release anytime soon, we shall manage it locally to remove dependency on "dotnet-eng" feed.

_Tool imported from dotnet/arcade_
(Developers should not notice any difference in build and execution)

I did not import these new projects in MDS Solution since they're not to be actively worked upon, let me know for any suggestions/opinions.

cc @David-Engel

### CODE REVIEW

File: tools/GenAPI/Microsoft.Cci.Extensions/Experimental/TrimAPIMutator.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (karinazhou) [inline tools/GenAPI/Microsoft.Cci.Extensions/Experimental/TrimAPIMutator.cs:1]: Do we need this file?

File: tools/GenAPI/Microsoft.Cci.Extensions/Extensions/CustomAttributeProvider.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
+using System;
```
Comment (karinazhou) [inline tools/GenAPI/Microsoft.Cci.Extensions/Extensions/CustomAttributeProvider.cs:1]: Missing headings

File: tools/GenAPI/Microsoft.Cci.Extensions/Extensions/FakeCustomAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
```
Comment (karinazhou) [inline tools/GenAPI/Microsoft.Cci.Extensions/Extensions/FakeCustomAttribute.cs:1]: Missing headings

File: tools/GenAPI/Microsoft.Cci.Extensions/SRMetadataPEReaderCache.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
+using System;
```
Comment (karinazhou) [inline tools/GenAPI/Microsoft.Cci.Extensions/SRMetadataPEReaderCache.cs:1]: License

**Review by karinazhou — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

**Pull request merged**